components/layout/display_list_builder.rs

```diff
@@ -196,16 +198,31 @@ impl<'a> DisplayListBuildState<'a> {
         }
     }
 
+    fn update_overflow_for_stacking_context(&mut self, stacking_context: &mut StackingContext) {
+        if stacking_context.context_type != StackingContextType::Real {
+            return;
+        }
+
+        for child in self.stacking_context_children(stacking_context.id).iter_mut() {
+            if child.context_type == StackingContextType::Real {
+                // This child might be transformed, so we need to take into account
+                // its transformed overflow rect too, but at the correct position.
+                let overflow = child.overflow_rect_in_parent_space();
+                stacking_context.overflow = stacking_context.overflow.union(&overflow);
+            }
+        }
+    }
+
     fn to_display_list_for_stacking_context(&mut self,
                                             list: &mut Vec<DisplayItem>,
-                                            mut stacking_context: StackingContext,
+                                            stacking_context: StackingContext,
                                             scroll_root_stack: &mut Vec<ScrollRootId>) {
         let mut child_items = self.items.remove(&stacking_context.id).unwrap_or(Vec::new());
         child_items.sort_by(|a, b| a.base().section.cmp(&b.base().section));
         child_items.reverse();
 
         let mut child_stacking_contexts =
-            mem::replace(&mut stacking_context.children, Vec::new());
+            mem::replace(self.stacking_context_children(stacking_context.id), Vec::new());
```

Comment:
Same here, this doesn't need to create an entry if the stacking context has no children.